### PR TITLE
apic-extension GA release

### PR DESCRIPTION
**Pull request by blackchoey**
---

This PR include changes for apic-extension GA
* Refactored some commands per feedback, as well as fixed some bugs. Please refer the changelog for more details.
* Added test cases for the CLI

### Related command
az apic *

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: .github/workflows/CD.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+name: CD
```
Comment (blackchoey) [inline .github/workflows/CD.yml:1]: I will remove this file after resolved all the comments

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: CI
```
Comment (blackchoey) [inline .github/workflows/CI.yml:1]: I will remove this file after resolved all the comments

**Comment by blackchoey** [comment]
aaz PR: https://github.com/Azure/aaz/pull/508

File: src/apic-extension/azext_apic_extension/azext_metadata.json [hunk 3]
```diff
@@ -1,4 +1,4 @@
 {
-    "azext.isPreview": true,
+    "azext.isPreview": false,
```
Comment (necusjz) [inline src/apic-extension/azext_apic_extension/azext_metadata.json:2]: from https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md, i found:
> For all stable extension releases, neither azext.isPreview nor azext.isExperimental can be added into module metadata.

**Comment by necusjz** [comment]
i saw there are a lot of "breaking changes" in https://github.com/Azure/azure-cli-extensions/pull/7677#issuecomment-2144543999, are they all expected?

**Comment by blackchoey** [comment]
> i saw there are a lot of "breaking changes" in [#7677 (comment)](https://github.com/Azure/azure-cli-extensions/pull/7677#issuecomment-2144543999), are they all expected?

@necusjz Yes, they're expected. We want to leverage the last chance to make some breaking changes to ensure the CLI is in the best shape. There're 3 kind of breaking changes:
1. Align the parameter naming with other Azure CLI commands. E.g. use `-n` instead `-s` to ask for resource name.
2. Mark some parameters as required, as they're not correctly marked before.
3. Adjust some command signatures per feedback.

File: src/apic-extension/azext_apic_extension/azext_metadata.json (hunk 3 above)
Comment (blackchoey) [reply src/apic-extension/azext_apic_extension/azext_metadata.json]: Thank you, removed the property.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**